### PR TITLE
[bug_5845] Added data from xml

### PR DESCRIPTION
**Pull request by pwodzinski**
https://bugzilla.stxnext.pl/show_bug.cgi?id=5845

### CODE REVIEW

File: buildout.cfg [hunk 1]
```diff
@@ -59,7 +60,7 @@ app = presence_analyzer#debug
 workers = 1
 spawn_if_under = 1
 max_requests = 0
-port = 5000
+port = 8089
```
Comment (dpolcyn) [inline buildout.cfg]: Do you have to change port here?

File: src/presence_analyzer/tests.py [hunk 2]
```diff
@@ -52,6 +57,39 @@ def test_api_users(self):
...
+        """
+        resp = self.client.get('/api/v2/users')
+        self.assertEqual(resp.status_code, 200)
+        self.assertEqual(resp.content_type, 'application/json')
+        data = json.loads(resp.data)
+        self.assertEqual(len(data), 2)
+        self.assertDictEqual(
+            data[0],
+            {
+                u'user_id': 176,
```
Comment (dpolcyn) [inline src/presence_analyzer/tests.py]: unicode

File: src/presence_analyzer/script.py [hunk 3]
```diff
@@ -111,3 +116,9 @@ def action_stop(dry_run=False):
         _serve('stop', dry_run=dry_run)
 
     werkzeug.script.run()
+
+
+def update_xml_web():
+    with open(DATA_XML, 'wb') as local:
+        web_file = urllib.urlopen(DATA_XML_WEB)
+        local.write(web_file.read())
```
Comment (dpolcyn) [inline src/presence_analyzer/script.py]: You can use ```urlretrieve```

File: src/presence_analyzer/utils.py [hunk 4]
```diff
@@ -71,6 +71,27 @@ def get_data():
     return data
 
 
+def parse_xml():
+    """
+    Extracs users' name and avatar from given xml document
+    """
+    data = {}
+    with open(app.config['DATA_XML'], 'r') as xmlfile:
```
Comment (mjelonek92) [inline src/presence_analyzer/utils.py]: You have to open in binary mode if you go this way.

Parse also accepts file path directly.
```etree.parse(app.config['DATA_XML'])``` should be prefectly fine.

http://lxml.de/tutorial.html#the-parse-function

File: src/presence_analyzer/views.py [hunk 5]
```diff
@@ -61,6 +62,33 @@ def users_view():
...
+    data = parse_xml()
+    return [
+        {'user_id': i, 'name': data[i]['name'], 'avatar': data[i]['avatar']}
+        for i in data.keys()
+    ]
+
+
+@app.route('/api/v2/presence/<int:user_id>', methods=['GET'])
+@jsonify
+def user_avatar(user_id):
```
Comment (mjelonek92) [inline src/presence_analyzer/views.py]: Why is this necessary? You already return a dict with all avatars in ```users_view2```!

File: src/presence_analyzer/views.py [hunk 6]
```diff
@@ -61,6 +62,33 @@ def users_view():
...
+    """
+    Returns total presence time of given user grouped by weekday.
```
Comment (mjelonek92) [inline src/presence_analyzer/views.py]: ?

File: src/presence_analyzer/utils.py [hunk 7]
```diff
@@ -71,6 +71,27 @@ def get_data():
     return data
 
 
+def parse_xml():
+    """
+    Extracs users' name and avatar from given xml document
```
Comment (dpolcyn) [inline src/presence_analyzer/utils.py]: I think you meant ```Extract```

Comment (dpolcyn) [inline src/presence_analyzer/utils.py]: Full stop on end of docstring.

File: src/presence_analyzer/views.py [hunk 8]
```diff
@@ -61,6 +62,33 @@ def users_view():
...
+@app.route('/api/v2/users', methods=['GET'])
+@jsonify
+def users_view2():
+    """
+    Users listing for dropdown.
+    """
+    data = parse_xml()
+    return [
+        {'user_id': i, 'name': data[i]['name'], 'avatar': data[i]['avatar']}
+        for i in data.keys()
```
Comment (mjelonek92) [inline src/presence_analyzer/views.py]: What is ```i```? Use names that are explanatory. In this case ```user_id``` would be more fitting.

File: src/presence_analyzer/static/js/weekday.js [hunk 9]
```diff
@@ -13,10 +13,21 @@ google.load("visualization", "1", {packages:["corechart"], 'language': 'en'});
         });
         $('#user_id').change(function() {
             var selected_user = $("#user_id").val(),
-                chart_div = $('#chart_div');
+                chart_div = $('#chart_div'),
+                userPhoto = $("#userPhoto").val('')
+                console.log(selected_user)
```
Comment (dpolcyn) [inline src/presence_analyzer/static/js/weekday.js]: You don't need this.

File: src/presence_analyzer/static/js/weekday.js [hunk 10]
```diff
@@ -25,6 +36,7 @@ google.load("visualization", "1", {packages:["corechart"], 'language': 'en'});
                     var chart = new google.visualization.PieChart(chart_div[0]);
                     chart.draw(data, options);
                 });
+
```
Comment (dpolcyn) [inline src/presence_analyzer/static/js/weekday.js]: You don't need this enter.

**Comment by dpolcyn** [comment]
Please change 'loading gif' with information, if data for specified user doesn't exist.

File: src/presence_analyzer/utils.py [hunk 11]
```diff
@@ -71,6 +73,26 @@ def get_data():
...
+    data = {}
+    tree = etree.parse(app.config['DATA_XML'])
+    root = tree.getroot()
+    host = root.findtext('./server/host')
+    port = root.findtext('./server/port')
+    protocol = root.findtext('./server/protocol')
+    serv = ('{}://{}:{}'.format(protocol, host, port))
+
+    for user in root.findall('./users/user'):
+        avatar = serv + user.find('avatar').text
```
Comment (dpolcyn) [inline src/presence_analyzer/utils.py]: You shouldn'tuse ```+``` for merging strings.

File: src/presence_analyzer/utils.py [hunk 12]
```diff
@@ -71,6 +73,26 @@ def get_data():
...
+    """
+    Extracts users' name and avatar from given xml document.
+    """
+    data = {}
+    tree = etree.parse(app.config['DATA_XML'])
+    root = tree.getroot()
+    host = root.findtext('./server/host')
+    port = root.findtext('./server/port')
+    protocol = root.findtext('./server/protocol')
+    serv = ('{}://{}:{}'.format(protocol, host, port))
```
Comment (dpolcyn) [inline src/presence_analyzer/utils.py]: You can do like this:
```
serv = ('{protocol}://{host}:{port}'.format(
    protocol=root.findtext('./server/protocol'),
    host = root.findtext('./server/host'),
    port = root.findtext('./server/port')
))

File: src/presence_analyzer/static/js/start_end.js [hunk 13]
```diff
@@ -33,12 +41,15 @@ google.load("visualization", "1", {packages:["corechart", "timeline"], 'language
...
                     chart_div.show();
                     loading.hide();
                     var chart = new google.visualization.Timeline(chart_div[0]);
                     chart.draw(data, options);
-                });
+                }).error(function() {
+                    loading.hide();
+                    chart_div.show();
+                    $('#chart_div').text("No information about user in data base!");
+                    });
```
Comment (dpolcyn) [inline src/presence_analyzer/static/js/start_end.js]: Too many indentations.
